test/python/test_roles_query_params_api.py

```diff
@@ -0,0 +1,507 @@
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.bad_auth_api.show_role(self.account, 'user', 'admin', graph='')
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_show_graph_404(self):
+        """Test case for show_role 404 response with graph query param"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.api.show_role(self.account, 'user', 'nonexist', graph='')
+
+        self.assertEqual(context.exception.status, 404)
+
+    # Test combinations of optional query parameters when getting role information
+
+    def test_parameter_combos_a(self):
+        """Test Conjur's response to being given all optional parameters in a single request
+        Conjur responds with `graph` results ONLY
+        """
+        details, status, _ = self.api.show_role_with_http_info(
+            self.account,
+            'user',
+            'admin',
+            all='',
+            memberships='',
+            members='',
+            graph=''
+        )
+
+        self.assertEqual(status, 200)
+
+        for i in details:
+            self.assertIn('parent', i)
+            self.assertIn('child', i)
+
+    def test_parameter_combos_b(self):
+        """Test Conjur's response to being given all optional parameters besides `graph`
+        Conjur responses with `all` results ONLY
+        """
+        details, status, _ = self.api.show_role_with_http_info(
+            self.account,
+            'user',
+            'admin',
+            all='',
+            memberships='',
+            members=''
+        )
+
+        target_details = [
+            self.ROOT_ID,
+            self.LAYER_ID,
+            self.ANOTHER_GROUP_ID,
+            self.ALICE_ID,
+            self.USER_GROUP_ID,
+            self.ADMIN_ID,
+            self.BOB_ID
+        ]
+
+        self.assertEqual(status, 200)
+        for i in target_details:
+            # This will throw an error causing the test to fail if
+            # i is not present in details
+            details.remove(i)
+
+        self.assertEqual(len(details), 0)
+
+    def test_parameter_combos_c(self):
+        """Test Conjur's response to being given both `members` and `memberships`
+        Conjur response with `memberships` results ONLY
+        """
+        self.add_user_to_group('bob')
+
+        details, status, _ = self.api.show_role_with_http_info(
+            self.account,
+            'user',
+            'bob',
+            memberships='',
+            members=''
+        )
+
+        target_details = {
+                'admin_option': False,
+                'ownership': False,
+                'role': self.USER_GROUP_ID,
+                'member': self.BOB_ID
+            }
+
+        self.assertEqual(status, 200)
+        self.assertEqual(len(details), 1)
+        self.assertEqual(details[0], target_details)
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
This is a good start, but this is still a very large file and I think it could be split at least one more time. It might help to also create a couple packages for the split out files